### PR TITLE
SPU: Inline and batch MFC list transfers

### PR DESCRIPTION
**Pull request by elad335**
Reduces overall CPU profiling load of this function from 9.6% to 6.8% and grants me about 2 fps in Sly 4. ~~This has a bit more effect when disabling Atomic RSX FIFO although it's partially active without it.~~

### CODE REVIEW

File: rpcs3/Emu/Cell/SPUThread.cpp [hunk 1]
```diff
@@ -2688,79 +2688,540 @@ bool spu_thread::do_dma_check(const spu_mfc_cmd& args)
...
+					// Assume success, prepare the next elements
+					arg_lsa += fetch_size * utils::align<u32>(s_size, 16);
+					item_ptr += fetch_size;
+					arg_size -= fetch_size * 8;
+
+					// Type which is friendly for fused address calculations
+					constexpr usz _128 = 128;
+
+					// Force constexpr std::max
+#define mov_t(type, index, _ea) { const usz ea = _ea; *reinterpret_cast<type*>(dst + index * std::integral_constant<u64, std::max<u64>(sizeof(type), sizeof(v128))>::value + (ea & 0xf)) = *reinterpret_cast<const type*>(src + ea); } void()
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/SPUThread.cpp:2777]: Use ALL_CAPS style for macro

File: rpcs3/Emu/Memory/vm_locking.h [hunk 2]
```diff
@@ -50,50 +50,22 @@ namespace vm
...
 		#ifndef _MSC_VER
 		__asm__(""); // Tiny barrier
 		#endif
 
-		u64 lock_addr = static_cast<u32>(lock_val); // -> u64
-		u32 lock_size = static_cast<u32>(lock_val << range_bits >> (32 + range_bits));
-
-		u64 addr = begin;
-
-		// Optimization: if range_locked is not used, the addr check will always pass
```
Comment (Nekotekina) [inline rpcs3/Emu/Memory/vm_locking.h:72]: Where did this code go?

Comment (elad335) [reply rpcs3/Emu/Memory/vm_locking.h:72]: The complete check is now in the function so register limit won't be reached, most of the time  it's unlocked so the initial check suffices. I've optimized the underlying function for it.

File: rpcs3/Emu/Cell/SPUThread.cpp (hunk 1 above)
Comment (elad335) [reply rpcs3/Emu/Cell/SPUThread.cpp]: Done,

**Comment by elad335** [comment]
Enabled fully the optimization for Atomic RSX FIFO which makes the perf boost more significant yet stability hasn't been sacrificed.

**Pull request merged**